openedx/openedx-platform
# fix: enroll users when added to a course team role

## Description

The course team management section under Instructor > Membership tab
allows users to be added a role even if are not enrolled in the course.
This behaviour does not match the help text displayed in the section.

This PR updates modify_access api to enrolls user if they are not enrolled
after adding them to a role as well as changes the help text to reflect
actual changes.

**Inconsistencies in studio and lms interface to add Admin/Staff users found by @kaustavb12**
Currently there are two ways to add an user as a course staff/admin. Either through the instructor dashboard in LMS, or through the Studio. 
In Studio, if an unenrolled user is added as the course staff, the user is auto enrolled, even though the message in the Studio Course team page says otherwise (It says "All course team members can access content in Studio, the LMS, and Insights, but are not automatically enrolled in the course") :

However, in LMS, if an unenrolled user is added as the course staff, the user is NOT auto-enrolled. Although the message there says that only enrolled users can have course team roles.

_This PR is updates LMS instructor dashboard behaviour to match with studio as well as studio text to reflect the real changes._

#32436 Is fixing the API for forum related roles as there is no inconsistency in case of forum related roles.

## Supporting information

`Private-ref`: [BB-7609](https://tasks.opencraft.com/browse/BB-7609)

## Testing instructions

* Setup master devstack or tutor locally.
* Register two users, one of which should have staff access.
* Using the staff user, enroll to a course.
* Under `Instructor > Membership` tab, go to `Course Team Management` section.
* Without the changes in this PR, you will be able to add the other user (who did not enroll in this course) to any role.
* With the changes in this PR, it will raise an error with appropriate msg.

![image](https://github.com/openedx/edx-platform/assets/10894099/c7f56449-1b8e-4d71-a587-fcbf799950e1)

## Other information

**A point of discussion**: _Should we revoke user's course related roles when they unroll from a course?_
